Microsoft.NET.Build.Containers.UnitTests/DockerDaemonAvailableUtils.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace Microsoft.NET.Build.Containers.UnitTests;
+
+public class DockerDaemonAvailableTheoryAttribute : TheoryAttribute
+{
+    private static bool IsDaemonAvailable = new LocalDocker(Console.WriteLine).IsAvailable();
+    public DockerDaemonAvailableTheoryAttribute()
+    {
+        if (!IsDaemonAvailable)
```

Comment:
This is a bit concerning for CI scenarios.

Skipped tests do not fail build, and change in environment which makes docker unavailable for some reason will be hide the failure.

imho, it would be good to check also if we are running CI or not.

Likely it would be easier to do it after move to arcade.

Reply:
I haven't been able to determine if AzDO sets the conventional `CI` environment variable to `true`, but we should be able to find _something_ to detect pretty easily and then fail fast :+1: